cohorts/load.py

```diff
@@ -521,7 +526,77 @@ def _load_single_patient_variants(self, patient, variant_type, merge_type):
 
         return merged_variants
 
-    def load_effects(self, patients=None, only_nonsynonymous=False, variant_type="snv", merge_type="union"):
+    def load_polyphen_annotations(self, database_file):
+        """Load a dataframe containing polyphen2 annotations for all variants
+
+        Parameters
+        ----------
+        database_file : string, sqlite
+            Path to the WHESS/Polyphen2 SQLite database.
+            Can be downloaded and bunzip2'ed from http://bit.ly/208mlIU
+
+        Returns
+        -------
+        annotations
+            Dictionary of patient_id to a DataFrame that contains annotations
+        """
+        patient_annotations = {}
+        for patient in self:
+            annotations = self._load_single_patient_polyphen(patient, database_file)
+            patient_annotations[patient.id] = annotations
+        return patient_annotations
+
+    def _load_single_patient_polyphen(self, patient, database_file):
+        cache_name = self.cache_names["polyphen"]
+        cached_file_name = "polyphen-annotations.csv"
+        cached = self.load_from_cache(cache_name, patient.id, cached_file_name)
+        if cached is not None:
+            return cached
+
+        engine = create_engine('sqlite:///{}'.format(database_file))
+        conn = engine.connect()
+
+        pvariants = self._load_single_patient_variants(patient,
+                                                       variant_type="snv",
+                                                       merge_type="union")
+        df = pd.DataFrame(columns=['chrom', 'pos', 'ref', 'alt',
+                                   'annotation_found', 'gene', 'protein',
+                                   'aa_change', 'hvar_pred', 'hvar_prob',
+                                   'hdiv_pred', 'hdiv_prob'])
+        for v in pvariants:
+            chrom = "chr{}".format(v.contig)
+            pos = v.start
+            ref = v.ref
+            alt = v.alt
+            annotation = vap.annotate_variant(conn, chrom, pos, ref, alt)
+            datum = {'chrom': chrom,
```

Comment:
Instead of initializing everything to `None`, how about either `datum["aa_change"] = annotation.aa_change if annotation else none` or `dataum["aa_change"] = getattr(annotation, "aa_change", None)`?

In fact, using `getattr`, you could simplify to:

```
for attr ["annotation_found", "gene", ...]:
   datum[attr] = getattr(annotation, attr, None)
```